### DISCUSSION TITLE
Trigger Dagster sensor/schedule via DagsterInstance.ephemeral()

### DISCUSSION THREAD

**Discussion by CHB-BH** [dagster-io/dagster#19462] [Q&A]
Hi Dagster Team, is it possible to trigger a Dagster sensor/schedule via local `DagsterInstance.ephemeral()`? So that the triggered run automatically contains tags like:

```
# for schedule
{"dagster/schedule_name": "sample_schedule"}

# for sensor
{"dagster/sensor_name": "sample_sensor"}
```

We are writing simple unittests to make sure our sensors/schedules are configured correctly. Similar to how we use `job.execute_in_process()` to evaluate it outside of Dagster environment (for testing purpose).

e.g. in the sample code below, we can get the RunRequest from `evaluate_tick()`, but we are not sure how to trigger a run using those RunRequests. 

```python
@op
def sample_op():
    yield Output(None)

@job
def sample_job():
    sample_op()

@sensor(job=sample_job)
def sample_sensor():
    yield RunRequest()

instance: DagsterInstance = DagsterInstance.ephemeral()
context: SensorEvaluationContext = build_sensor_context(instance=instance)
data: SensorExecutionData = sample_sensor.evaluate_tick(context)
for run_request in data.run_requests:
    print(run_request)
```

Thanks.

**Comment by alangenfeld** [comment]
Unfortunately theres no existing code that you can use for this, but you should be able to cobble something decent together. 

You can grab things like `run_config` and `tags` off of  `RunRequest` 
https://docs.dagster.io/_apidocs/schedules-sensors#dagster.RunRequest
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/run_request.py#L109

and you can get the sensor/schedule tags that get set from these methods on `DagsterRun`
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/storage/dagster_run.py#L427-L433
https://docs.dagster.io/_apidocs/internals#dagster.DagsterRun

**Reply by CHB-BH** [reply]
Thanks for suggestions! We will explore them.

On the other hand, upon closer inspection, we noticed that schedule-evaluated-tick contains the tag `{'dagster/schedule_name': 'sample_schedule'}` correctly.

However, the sensor-evaluated-tick does NOT contain the tag `{'dagster/sensor_name': 'sample_sensor'}`

By any chance is this a bug? Or is this bug fixed in a future Dagster version (we are on 1.5.6)?

Or is this the expected design decision on Dagster's end (and the tag for sensor is added in another part of Dagster's code)?

Thanks.

```python

@op
def sample_op():
    yield Output(None)


@job
def sample_job():
    sample_op()


@sensor(job=sample_job)
def sample_sensor():
    yield RunRequest()


@schedule(cron_schedule="* * * * *", job=sample_job)
def sample_schedule():
    yield RunRequest()


instance: DagsterInstance = DagsterInstance.ephemeral()
sensor_context: SensorEvaluationContext = build_sensor_context(instance=instance)
sensor_data: SensorExecutionData = sample_sensor.evaluate_tick(context=sensor_context)

schedule_context: ScheduleEvaluationContext = build_schedule_context(instance=instance)
schedule_data: ScheduleExecutionData = sample_schedule.evaluate_tick(context=schedule_context)


# RunRequest(run_key=None, run_config={}, tags={}, job_name=None, asset_selection=None, stale_assets_only=False, partition_key=None)
print(sensor_data.run_requests[0])  # missing tags

# RunRequest(run_key=None, run_config={}, tags={'dagster/schedule_name': 'sample_schedule'}, job_name=None, asset_selection=None, stale_assets_only=False, partition_key=None)
print(schedule_data.run_requests[0])  # tags OK

```
